system/jlib/jptree.cpp

```diff
@@ -1046,67 +1046,67 @@ void PTree::resolveParentChild(const char *xpath, IPropertyTree *&parent, IPrope
         Owned<IPropertyTreeIterator> pathIter = getElements(path);
         if (!pathIter->first())
             throw MakeIPTException(-1, "resolveParentChild: path not found %s", xpath);
-
-        IPropertyTree *currentPath = NULL;
-        bool multiplePaths = false;
-        bool multipleChildMatches = false;
-        for (;;)
+        if (this != &pathIter->query())
```

Comment:
NB: diff make it look like a much bigger change than it is.

In fact this line is the significant change/addition, with the pre-exisitng code below it indented into a block with a return at the end of it, falling through to what used to be the else clause at old line 1091

Reply:
This could do with a comment explaining what this test and branch are doing.  The change looks logical, but it is very hard to work out if it is correct.

Reply:
agree, will add a comment